### PR TITLE
systemctl: make show/status honour --state and --type

### LINKED ISSUE #23207: systemctl status --failed, please?
To show more details about failed units and to avoid typing lengthy service names, something like 
```
systemctl status --failed
```
would be very nice. Currently (250.4) the "--failed" appears to be ignored (without usage hint).

Just a suggestion, of course.

### PR DESCRIPTION
**Pull request by keszybz**
This makes the interface more flexible, by allowing the same filtering
for show and status as is done for list-units.

Fixes #23207.

### CODE REVIEW

**Comment by bluca** [comment]
Do the shell completions need a tweak for this?

**Comment by keszybz** [comment]
> Do the shell completions need a tweak for this?

I don't think so, because it's just extending coverage of existing options to new verbs. I don't think any of our completions are smart enough to know which verbs care about which options.

File: src/systemctl/systemctl-show.c [hunk 1]
```diff
@@ -2196,17 +2197,28 @@ int verb_show(int argc, char *argv[], void *userdata) {
...
+                        if (show_mode == SYSTEMCTL_SHOW_PROPERTIES)
+                                r = show_one(bus, "/org/freedesktop/systemd1", NULL, show_mode, &new_line, &ellipsized);
+                        else
+                                r = show_system_status(bus);
+                        if (r < 0)
+                                return r;
 
-                show_system_status(bus);
-                new_line = true;
+                        new_line = true;
```
Comment (daandemeyer) [inline src/systemctl/systemctl-show.c:2217]: Is it intended that we also always show a newline when we're following the `SYSTEMCTL_SHOW_PROPERTIES` logic path now? `show_one` seems to set `new_line` itself so it's weird that we just override it here now.

**Review by bluca — APPROVED** [review]

File: src/systemctl/systemctl-show.c (hunk 1 above)
Comment (keszybz) [reply src/systemctl/systemctl-show.c:2217]: Yes. The newline separates the system status from the per-unit statuses. (And the per-unit statuses are also separated by an empty line.)

**Comment by keszybz** [comment]
?

**Comment by yuwata** [comment]
Maybe better to add a test?

**Comment by keszybz** [comment]
In principle I'd agree, but we have exactly 0 tests for `--failed` or `list-units`, except for incidental use in other tests. We should add a full test for systemctl that at least verifies the main aspects of operation… But I don't want to be on the hook for that for this relatively simple pull request.

**Comment by yuwata** [comment]
OK. Let's add tests later in another PR.

**Pull request merged**